app/src/main/java/com/github/swent/echo/compose/components/ListDrawer.kt

```diff
@@ -73,22 +75,23 @@ fun ListDrawer(
                 .testTag("list_drawer")
     ) {
         // Iterate over the list of events and display them
-        items(eventsList.size) { index ->
-            val event = eventsList[index]
-            val canModifyEvent = event.creator.userId == userId
+        eventsList.forEach { event ->
+            item {
+                val canModifyEvent = event.creator.userId == userId
 
-            EventListItem(
-                event = event,
-                selectedEvent = selectedEvent,
-                isOnline = isOnline,
-                refreshEvents = refreshEvents,
-                viewOnMap = viewOnMap,
-                modify = modify,
-                onTagPressed = onTagPressed,
-                canModifyEvent = canModifyEvent,
-                distance = distances?.get(index),
-            )
-            Spacer(modifier = Modifier.height(12.dp))
+                EventListItem(
+                    event = event,
+                    selectedEvent = selectedEvent,
+                    isOnline = isOnline,
+                    refreshEvents = refreshEvents,
+                    viewOnMap = viewOnMap,
+                    modify = modify,
+                    onTagPressed = onTagPressed,
+                    canModifyEvent = canModifyEvent,
+                    distance = userLocation?.let { event.location.toLatLng().distanceTo(it) },
+                )
+                Spacer(modifier = Modifier.height(12.dp))
```

Comment:
Similarly for the height value, could be extracted into a variable to b e able to make modifications easily by someone else in the future